hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/DefaultRecordStore.java

```diff
@@ -773,34 +773,28 @@ public boolean merge(MapMergeTypes mergingEntry, SplitBrainMergePolicy<Data, Map
             storage.put(key, record);
             eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
                     key, null, record.getValue());
-        } else {
-            oldValue = record.getValue();
-            MapMergeTypes existingEntry = createMergingEntry(serializationService, record);
-            newValue = mergePolicy.merge(mergingEntry, existingEntry);
-            // existing entry will be removed
-            if (newValue == null) {
-                removeIndex(record);
-                mapDataStore.remove(key, now);
-                onStore(record);
-                eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(),
-                        partitionId, key, oldValue, null);
-                storage.removeRecord(record);
-                return true;
-            }
-            if (newValue == mergingEntry.getValue()) {
-                mergeRecordExpiration(record, mergingEntry);
-            }
-            // same with the existing entry so no need to map-store etc operations.
-            if (recordComparator.isEqual(newValue, oldValue)) {
-                return true;
-            }
-            newValue = mapDataStore.add(key, newValue, now);
+            saveIndex(record, null);
+            return newValue != null;
+        } else if (newValue == null) {
+            mapDataStore.remove(key, now);
             onStore(record);
             eventJournal.writeUpdateEvent(mapContainer.getEventJournalConfig(), mapContainer.getObjectNamespace(), partitionId,
```

Comment:
This is probably wrong, it should be `eventJournal.writeRemoveEvent` (since we are removing the entry).

Reply:
Oh, good one. Does the `writeRemoveEvent()` method get the `oldValue` as parameter? For me that's a bit unclear from the JavaDoc. It just says `the entry value`.

Reply:
Yes, you can pass in the value being deleted.